test/e2e/apps/disruption.go

```diff
@@ -76,6 +109,34 @@ var _ = SIGDescribe("DisruptionController", func() {
 		framework.ExpectNoError(err)
 	})
 
+	ginkgo.It("should update/patch PodDisruptionBudget status", func() {
+		createPDBMinAvailableOrDie(cs, ns, defaultName, intstr.FromInt(1), defaultLabels)
+		createPodsOrDie(cs, ns, 3)
+		waitForPodsOrDie(cs, ns, 3)
+		pod, _ := locateRunningPod(cs, ns)
+		newDisruptionsAllowed := int32(0)
+
+		ginkgo.By("Updating PodDisruptionBudget status")
+		updated, _ := updatePDBOrDie(cs, ns, defaultName, func(old *policyv1beta1.PodDisruptionBudget) *policyv1beta1.PodDisruptionBudget {
+			old.Status.DisruptionsAllowed = newDisruptionsAllowed
+			old.Status.DisruptedPods = make(map[string]metav1.Time)
+			old.Status.DisruptedPods[pod.Name] = metav1.NewTime(time.Now())
+			return old
+		}, cs.PolicyV1beta1().PodDisruptionBudgets(ns).UpdateStatus)
+		framework.ExpectEqual(updated.Status.DisruptionsAllowed, newDisruptionsAllowed, "Expecting the DisruptionsAllowed's DisruptionsAllowed to be updated to %d", newDisruptionsAllowed)
+		updated, _ = getPDBStatusOrDie(dc, ns, defaultName)
+		framework.ExpectHaveKey(updated.Status.DisruptedPods, pod.Name, "Expecting the DisruptedPods have %s", pod.Name)
```

Comment:
Why do we check `DisruptionsAllowed` on the PDB returned from `updatePDBOrDie` but we fetch the PDB again before checking the `DisruptedPods`?